mobile-widgets/qmlmanager.cpp

```diff
@@ -869,13 +869,11 @@ void QMLManager::commitChanges(QString diveId, QString date, QString location, Q
 		d->suit = strdup(qPrintable(suit));
 	}
 	if (myDive->buddy() != buddy) {
-		if (myDive->buddy().contains(",")) {
-			if (!buddy.contains(tr("Multiple Buddies"))) {
-				diveChanged = true;
-				free(d->buddy);
-				d->buddy = strdup(qPrintable(buddy));
-			}
-		} else {
+		if (buddy.contains(",") && !buddy.contains(QRegExp("(,\\s)"))){
```

Comment:
See my comment in the overall review above. We need to convert "\\s*,\\s*" into ", " for this all to work as intended. That could be done with a single QString replace that uses exactly that regex and replaces it every time it finds it. My comment simply pointed out that the way that you wrote the code, we could end up with extra spaces in front of buddy names and then you'd suddenly have "John" and " John" in your list of buddies - and that would be confusing.